cmd/device-hub/cmd_server.go

```diff
@@ -29,6 +30,12 @@ var serverCommand = &cobra.Command{
 
 		var dataImpl store.Storer
 
+		// Load Config File if is enabled
+		if _config.ConfigFile {
+			log.Println("Overriding all settings with config file")
+			_config.AddConfigFile(_config.ConfigPath)
+		}
+
```

Comment:
The first couldn't be made without affect the precedence order we want, but as the server instance is the only that require that configuration, I don't think it could be a problem to have that stanza out of init.